pyprecag/processing.py

```diff
@@ -2537,3 +2401,222 @@ def ttest_analysis(points_geodataframe, points_crs, values_raster, out_folder,
             dur=timedelta(seconds=time.time() - loop_time)))
 
     return df_table
+
+
+def persistor_all_years(raster_files, output_tif, greater_than, target_percentage):
+    """Determine the performance persistence of yield by across multiple years as described in
+    Bramley and Hamilton (2005)
+
+    The "All Years" method identifies cells for each raster within the target percentage above or
+    below the mean. These are then summed to create a count representing the number of years
+    the criteria was met.
+
+     All input rasters MUST overlap and have the same coordinate system and pixel size.
+
+     If a path is omited from output_tif it will be created in your temp folder.
+
+    References:
+        Bramley RGV, Hamilton RP (2005) Understanding variability in winegrape production systems
+        1. Within vineyard variation in yield over several vintages. Australian Journal Of Grape And
+        Wine Research 10, 32-45. doi:10.1111/j.1755-0238.2004.tb00006.x.
+
+    Args:
+        raster_files (List[str]): List of rasters to use as inputs
+        output_tif (str): Output TIF file
+        greater_than (bool): if true test above (gt) the mean
+        target_percentage (int): the percent variation either above/below the mean. This should be
+                   a integer between -50 and 50
+
+    Returns:
+        str: The output tif name
+    """
+
+    if not isinstance(target_percentage, int):
+        raise TypeError('target_percentage must an integer between -50 to 50.')
+
+    if target_percentage < -50 or target_percentage > 50:
+        raise ValueError('target_percentage must an integer between -50 to 50.')
+
+    if not isinstance(greater_than, (int, bool)):
+        raise TypeError('greater_than must be boolean.')
+
+    if output_tif is not None and not os.path.isabs(output_tif):
+        output_tif = os.path.join(TEMPDIR, output_tif)
+
+    if not isinstance(raster_files, list):
+        raise TypeError('Invalid Type: raster_files should be a list')
+
+    if len(raster_files) == 0:
+        raise TypeError('Invalid Type: Empty list of raster files')
+
+    start_time = time.time()
+
+    # combine and align grid and slip to common area
+    image_combined, _ = stack_and_clip_rasters(raster_files, use_common=True,
+                                               output_tif='allyrs_combined.tif')
+
+    with rasterio.open(image_combined, 'r') as src:
+        kwargs = src.meta.copy()
+
+        for i, band in enumerate(src.read(masked=True), start=1):
+            cutoff = np.nanmean(band) + (np.nanmean(band) * (target_percentage / 100.00))
+            band_new = band.copy()
+
+            # classifiy bands
```

Comment:
Strictly speaking, this should be slightly different logic as we are testing for >= when the function says >.  And the idea of 'not greater_than' is slightly different from that of 'less than'.

if (greater_than == True) 
    # set to 1 for > cutoff, and 0 otherwise
    band_new[np.where(band > cutoff)] = 1
    band_new[np.where(band <= cutoff)]  = 0
else
    # the 'less than' case 
    #  set to 1 for < cutoff and 0 otherwise
    band_new[np.where(band < cutoff)] = 1
    band_new[np.where(band >= cutoff)]  = 0

Reply:
This is the same as the original code where we used **'>'** and **'<'**. Ive just converted this to a boolean.
Then it reclassifies using  2 9 0 logic  which as per documentation means assign 0 to values **greater than** 2 and  **less than or equal** to 9 which I have also replicated. 
I can change the variable name if necessary, but this also matches the gui's wording.